build/push-release-tarball.sh

```diff
@@ -5,12 +5,11 @@ set -euxo pipefail
 # This file uses `bash` and not `sh` due to the `time` builtin (the external
 # `time` is not available on CircleCI).
 
-BUCKET_NAME="binaries.cockroachdb.com"
```

Comment:
I moved this to prevent duplication of the bucket name more than anything else; it wasn't configurable in the original version. Want me to make configurable, though?